src/packages/cli/src/detach.ts

```diff
@@ -0,0 +1,250 @@
+import { fork } from "child_process";
+import createInstanceName from "./process-name";
+import envPaths from "env-paths";
+import psList from "@trufflesuite/ps-list";
+import { readFile, rm, mkdir, readdir, writeFile } from "fs/promises";
+import path from "path";
+import { FlavorName } from "@ganache/flavors";
+
+export type DetachedInstance = {
+  instanceName: string;
+  pid: number;
+  startTime: number;
+  host: string;
+  port: number;
+  flavor: FlavorName;
+  cmd: string;
+  version: string;
+};
+
+const READY_MESSAGE = "ready";
+
+const START_ERROR = "An error ocurred spawning a detached instance of Ganache:";
+
+const dataPath = new Promise<string>(async (resolve, reject) => {
+  const path = envPaths(`Ganache/instances`).data;
+  // it seems strange to call mkdir blindly, but this is to avoid a race
+  // condition where the directory is modified between checking it's existence,
+  // and creating it see https://nodejs.org/api/fs.html#fsexistspath-callback
+  try {
+    await mkdir(path, { recursive: true });
+  } catch (err) {
+    if ((err as { code: string }).code === "EEXIST") {
+      resolve(path);
+    }
+    reject(path);
+  }
+  resolve(path);
```

Comment:
The directory should probably be created only at the moment in which we attempt to write a file to the directory. It doesn't need to exist for reads.

This has a nice benefit of not requiring `await dataPath` in these case when we just want to _read_ or delete from the directory, and it also doesn't cause a random filesystem call on all Ganache start ups (we don't need to proactively create the directory, so I don't think we should).